tests/unit/utils/serialize/test_yaml.py

```diff
@@ -12,3 +16,17 @@ def test_parse_yaml_duplicate_key_error():
     """
     with pytest.raises(YAMLError, match='found duplicate key "mykey"'):
         parse_yaml(text, "mypath")
+
+
+def test_parse_yaml_invalid_unicode():
```

Comment:
We use `tmp_dir` fixture for creating a temporary directory and generate the file using `tmp_dir.gen` here, similar to:
https://github.com/iterative/dvc/blob/d6afd26beae7c4d9b4192abfb4a37506aa2186e4/tests/func/metrics/test_diff.py#L128-L129

[tests/dir_helpers.py](https://github.com/iterative/dvc/blob/1d3524f14403363046048ea4aa348042fa4006f4/tests/dir_helpers.py) has some docstring on how to use it.

Reply:
Thanks for pointing this out! I have updated the test!